### PR TITLE
Convert tests to Jest-TS

### PR DESCRIPTION
**Pull request by szmarczak**
What I don't like about this approach is that it brings a lot of config files.

### CODE REVIEW

**Comment by szmarczak** [comment]
Yep, they're just to get the build rolling. Will fix them.

File: .eslintrc [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+      "project": "tsconfig.eslint.json"
+    },
+    "overrides": [
+      {
+        "files": [
+          "src/**/*.ts"
+        ],
+        "rules": {
+          "import/extensions": 0
+        }
```
Comment (mnmkng) [inline .eslintrc:15]: This could be added to the eslint-config, no? I think I'm seeing this repeatedly.

File: src/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	"compilerOptions": {
+		"rootDir": "./",
+		"outDir": "../dist",
+		"composite": true
+	},
+	"include": [
+		".",
+		"./**/*.d.ts"
+	]
+}
```
Comment (mnmkng) [inline src/tsconfig.json:13]: @vladfrangu we'll replace both this and the `base` with the new tsconfig package, right?

File: tsconfig.base.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (mnmkng) [inline tsconfig.base.json:1]: Ok, now I see why @szmarczak is talking about "a lot of config files". Can we make this better somehow? This config will get out of sync between packages sooner or later.

**Review by mnmkng — COMMENTED** [review]
Yeah. A lot of config files. I generally don't mind them but can we do something about it to clean it up a bit?

File: .eslintrc (hunk 1 above)
Comment (szmarczak) [reply .eslintrc:15]: I just copied everything from `browser-pool`.

Comment (mnmkng) [reply .eslintrc:15]: Yeah, I know.

Comment (vladfrangu) [reply .eslintrc:15]: Pretty sure it could be moved there, yes

File: src/tsconfig.json (hunk 2 above)
Comment (mnmkng) [reply src/tsconfig.json:13]: Ok, keep us posted 👍

Comment (vladfrangu) [reply src/tsconfig.json:13]: You can see it in https://github.com/apify/apify-client-js/pull/185 actually 😄

**Pull request merged**